pkg/arvo/neo/cod/std/src/fil/types.txt

```diff
@@ -0,0 +1,467 @@
+
+## `aeon`
+
+Total version numbers and signature for a shrub. [How is the signature being used here?]
+
+```
++$  aeon  (pair ever oath)
+```
+
+## `axal`
+
+Fundamental node, recursive.
+
+This mold builder creates a representation of a node in Clay like an [arch](https://docs.urbit.org/language/hoon/reference/arvo#arch) (a file or a directory) or [axil](https://docs.urbit.org/language/hoon/reference/arvo#axil) (fundamental node), but the directory map contains more axals, so it contains the entire subtree rather than just one level.
+
+```
+++  axal
+  |$  [item]
+  [fil=(unit item) kid=(map iota $)]
+```
+
+## `band`
+
+Dependency map.
+
+```
++$  band  $+(band (map term fief))
+```
+
+The `term`s are arbitrary keywords defined in a shrub's `+deps` arm. When that shrub is created, the `conf` will have to include a map of all the required dependencies.
+
+## `bowl`
+
+`bowl:neo`, similar to `bowl:gall`.
+
+```
++$  bowl
+  $:  src=name
+      our=@p
+      were=pith
+      here=pith
+      now=@da
+      eny=@uvJ
+      deps=(map term (pair pith lore))
+      kids=lore
+  ==
+```
+
+- `src`: source ship and path [of what?].
+- `our`: our ship.
+- `were`: ???
+- `here`: location of this shrub.
+- `now`: current datetime.
+- `eny`: entropy.
+- `deps`: ???
+- `kids`: ???
+
+## `card`
+
+An instruction for a shrub, similar to a `card:agent:gall`.
+
+```
++$  card  (pair pith note)
+```
+
+The `pith` is the shrub this card will be sent to. The `note` will be of type `%make`, `%poke`, `%tomb`, or `%cull`.
+
+## `care`
+
+Perspective on a path.
+
+```
++$  care
+  $~  %x
+  $?  %x
+      %y
+      %z
+      %a
+      %b
+      %c
+  ==
+```
+
+The cares `%x`, `%y`, and `%z` appear throughout the system: you’ll use them while defining `kids` and `deps`, and state updates will be head-tagged with one of these cares.
+
+[not sure about %a, %b, %c; i know the correspond to x y z but not sure about usage]
+
+## `conf`
+
+Dependency configuration for a new shrub.
+
+```
++$  conf  (map term pith)
+```
+
+This can be included in a `%make` card when we want to pass dependencies into a new shrub.
+
+The `term`s are arbitrary tags the new shrub has defined, and the `pith`s are locations of the relevant shrubs whose state will be tracked for the new shrub to respond to.
+
+## `curb`
+
+Constraint on the state of a shrub.
+
+```
++$  curb
+  $~  [%pro %$]
+  $%  [%or p=(list curb)]
+      [%only p=stud]
+      [%rol p=stud q=curb]
+      [%not p=curb q=curb]
+      [%pro p=stud]
+      [%any ~]
+  ==
+```
+
+[explain each individually]
+
+[how can they be combined, why would you do that?]
+
+## `dare`
+
+A `%y` or `%z` care. (Think "dare is a downward care".)
+
+```
++$  dare  ?(%y %z)
+```
+
+Used while constraining descendants.
+
+## `deed`
+
+Whether or not a depdenency must already exist in order for this shrub to build.
```

Comment:
Is this "must already exist" or is this "must be provided in the `conf` in order for this shrub to build"?